builder/dockerfile/internals.go

```diff
@@ -51,11 +51,11 @@ func (b *Builder) commit(id string, autoCmd strslice.StrSlice, comment string) e
 
 	if id == "" {
 		cmd := b.runConfig.Cmd
-		if runtime.GOOS != "windows" {
-			b.runConfig.Cmd = strslice.StrSlice{"/bin/sh", "-c", "#(nop) " + comment}
-		} else {
-			b.runConfig.Cmd = strslice.StrSlice{"cmd", "/S /C", "REM (nop) " + comment}
-		}
+		// Since this is a no-op and only used for documentation, it doesn't matter
+		// that using # as the start of comment on Windows with the default shell of
+		// cmd is used. Keeping it as # avoids backwards compatibility issues or
+		// potential cache misses.
```

Comment:
This is a good point and we should add a testcase to make sure that the history of an image shows the user-defined shell in there instead of "sh -c".